Modules/Core/Common/test/itkAdaptorComparisonTest.cxx

```diff
@@ -210,63 +197,78 @@ itkAdaptorComparisonTest(int, char *[])
   // Time trials
 
   std::cout << "Speed of adaptor supporting iterator (for reference) \t";
-  start = clock();
-  AdaptorSupportedIteratorSpeed(scalar_image);
-  stop = clock();
-  adaptor_comp = stop - start;
-  std::cout << adaptor_comp << std::endl;
-
-  std::cout << "Speed of iterator that does not support adaptors (for reference) \t";
-  start = clock();
-  NoAdaptorSupportIteratorSpeed(scalar_image);
-  stop = clock();
-  no_adaptor_comp = stop - start;
-  std::cout << no_adaptor_comp << std::endl;
 
-  std::cout << "Modifying scalar image using adaptor iterator...\t";
-  start = clock();
-  AdaptorSupportedModifyScalars(scalar_image);
-  stop = clock();
-  std::cout << (stop - start) << "\t compensated = " << (stop - start) - adaptor_comp << std::endl;
-
-  std::cout << "Modifying scalar image using non-adaptor iterator...\t";
-  start = clock();
-  NoAdaptorSupportModifyScalars(scalar_image);
-  stop = clock();
-  std::cout << (stop - start) << "\t compensated = " << (stop - start) - no_adaptor_comp << std::endl;
-
-  std::cout << "Modifying vector image using adaptor iterator...\t";
-  start = clock();
-  AdaptorSupportedModifyVectors(vector_image);
-  stop = clock();
-  std::cout << (stop - start) << "\t compensated = " << (stop - start) - adaptor_comp << std::endl;
-
-  std::cout << "Modifying vector image using non-adaptor iterator...\t";
-  start = clock();
-  NoAdaptorSupportModifyVectors(vector_image);
-  stop = clock();
-  std::cout << (stop - start) << "\t compensated = " << (stop - start) - no_adaptor_comp << std::endl;
-
-  std::cout << "Modifying scalar image bypassing adaptor api using"
-            << " adaptor iterator...\t";
-  start = clock();
-  BypassAdaptorSupportModifyScalars(scalar_image);
-  stop = clock();
-  std::cout << (stop - start) << "\t compensated = " << (stop - start) - adaptor_comp << std::endl;
-
-  std::cout << "Modifying vector image bypassing adaptor api using"
-            << " non-adaptor iterator...\t";
-  start = clock();
-  BypassNoAdaptorSupportModifyVectors(vector_image);
-  stop = clock();
-  std::cout << (stop - start) << "\t compensated = " << (stop - start) - adaptor_comp << std::endl;
-
-  std::cout << "Modifying vector image bypassing adaptor api using"
-            << " adaptor iterator...\t";
-  start = clock();
-  BypassAdaptorSupportModifyVectors(vector_image);
-  stop = clock();
-  std::cout << (stop - start) << "\t compensated = " << (stop - start) - adaptor_comp << std::endl;
+  const clock_t adaptor_comp = [&]() -> auto
```

Comment:
`-> auto` is not required here, of course. A lambda returns `auto` by default. Honestly I didn't see this `-> auto` syntax before. It's no show-stopper to me, of course, it just looks unfamiliar to me.

Instead of the default capture `[&]`, I would rather capture scalar_image explicitly, by value:

```cpp
    const clock_t adaptor_comp = [scalar_image]
        ...
```
Following Scott Meyers Effective Modern C++ item "Avoid default capture modes". (But that's no showstopper to me either.)

Reply:
Suggestions implemented.